Shared/ModuleHandle.ps1

```diff
@@ -0,0 +1,98 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+<#
+.SYNOPSIS
+This script defines a function `Request-Module` that checks for the presence of specified PowerShell modules.
+If a module is not found, it attempts to install it on current user scope.
+The function accepts a list of module names and an optional minimum version for the modules.
+It returns a boolean indicating whether all specified modules are installed successfully.
+
+.PARAMETER Modules
+ Mandatory array of strings specifying the names of the modules to check and install if necessary.
+.PARAMETER MinModuleVersion
+ Optional parameter to specify the minimum version of the modules (default is null).
+
+.OUTPUTS
+bool. A boolean indicating whether all specified modules are installed successfully.
+
+.EXAMPLE
+$requestModule = Request-Module -Modules "ExchangeOnlineManagement"
+This example checks if the "ExchangeOnlineManagement" module is installed. If it is not found, the script attempts to install it.
+
+.EXAMPLE
+$requestModule = Request-Module -Modules "ExchangeOnlineManagement" -MinModuleVersion $MinModuleVersion
+This example checks if the "ExchangeOnlineManagement" module with a specified minimum version is installed. If it is not found, the script attempts to install it.
+#>
+
+. $PSScriptRoot\..\Shared\Confirm-Administrator.ps1
+
+function Request-Module {
+    [OutputType([bool])]
+    param (
+        [Parameter(Mandatory = $true)]
+        [string[]]$Module,
+        [Parameter(Mandatory = $false)]
+        [System.Version]$MinModuleVersion = $null,
+        [Parameter(Mandatory = $false)]
+        [switch]$InstallAllUsersIfNotAvailable
+    )
+
+    $noFoundError = $true
+
+    foreach ($m in $Module) {
+        Write-Verbose "Checking $m PowerShell Module"
+        $getParams = @{
+            Name        = $m
+            ErrorAction = 'SilentlyContinue'
```

Comment:
@iserrano76 should we change this to `ErrorAction = Stop` and put the `Get-InstalledModule` into a `try/catch`? If we fail to query the modules for whatever reason, we treat it the same as if the module was not found. I think we should change this behavior.

Reply:
If we are handling it properly, does it matter if we do the `try/catch` version?

Reply:
No, it shouldn't matter then. If I understand the code correctly, we're trying to install the module if `$installed` is `$null` or does not contain `$m` (module x). Installed would be `$null` in case that `Get-InstalledModule` fail for whatever reason. This would mean we're trying installing a module which might be already there. That's my understanding here. Please correct me if I'm wrong.